cpp/src/groupby/hash/groupby.cu

```diff
@@ -65,6 +65,15 @@ namespace detail {
 namespace hash {
 namespace {
 
+// TODO: replace it with `cuco::static_map`
+// https://github.com/rapidsai/cudf/issues/10401
+using map_type = concurrent_unordered_map<
+  cudf::size_type,
+  cudf::size_type,
```

Comment:
I notice that we never use map values. Can we use `char` as value type to save memory?

Reply:
@jrhemstad mentioned yesterday in cuco meeting that we cannot use `set` here but I don't recall why exactly.

Reply:
Also, using a `char` wouldn't save memory. The size of a slot would still be `8B` due to padding needed for performance.

Reply:
> We could use a set but we don't have one of those yet :)

the grand cuco refactoring we are expecting